### PR TITLE
Add option to specify custom NNAPI serializer

### PR DESCRIPTION
**Pull request by axitkhurana**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61025**
* #61024
* #61023
* #61022
* #61021
* #59529
* #59434

Summary:

To add serializer for custom ops we can subclass default serializer
and update ADDER_MAP

Test Plan:

* pytest test/test_nnapi.py::TestNNAPI for current serializer
* Custom serializers to be tested with custom ops

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29480745](https://our.internmc.facebook.com/intern/diff/D29480745)

### CODE REVIEW

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480745).

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480745).

File: torch/backends/_nnapi/prepare.py [hunk 1]
```diff
@@ -75,14 +76,16 @@ def forward(self, args: List[torch.Tensor]) -> List[torch.Tensor]:
...
     model = torch.jit.freeze(model)
 
     if isinstance(inputs, torch.Tensor):
         inputs = [inputs]
 
-    ser_model, used_weights, inp_mem_fmts, out_mem_fmts, shape_compute_lines, retval_count = serialize_model(model, inputs)
+    if serializer:
+        ser_model, used_weights, inp_mem_fmts, out_mem_fmts, shape_compute_lines, retval_count = serializer.serialize_model(model, inputs)
+    else:
+        ser_model, used_weights, inp_mem_fmts, out_mem_fmts, shape_compute_lines, retval_count = _NnapiSerializer(config=None).serialize_model(model, inputs)
```
Comment (dreiss) [inline torch/backends/_nnapi/prepare.py:88]:
```
serializer = serializer or _NnapiSerializer(config=None)
ser_model, used_weights, inp_mem_fmts, out_mem_fmts, shape_compute_lines, retval_count = serializer.serialize_model(model, inputs)
```

**Review by dreiss — APPROVED** [review]

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480745).

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480745).

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29480745).

**Pull request closed without merging**